status/status.go

```diff
@@ -89,9 +90,15 @@ func FromError(err error) (s *Status, ok bool) {
 	if err == nil {
 		return nil, true
 	}
-	var se interface{ GRPCStatus() *Status }
-	if errors.As(err, &se) {
-		return se.GRPCStatus(), true
+	type grpcstatus interface{ GRPCStatus() *Status }
+	if gs, ok := err.(grpcstatus); ok {
+		return gs.GRPCStatus(), true
+	}
+	var gs grpcstatus
+	if errors.As(err, &gs) {
+		p := gs.GRPCStatus().Proto()
+		p.Message = err.Error()
+		return status.FromProto(p), true
 	}
```

Comment:
Why do you use a proto intermediary here rather than just use New(/*code extracted here*/, err.Error())?

Reply:
Because otherwise the status details would be lost.

Reply:
Oh right